### PR TITLE
[FIX] website_crm : impossible to add a crm form

### PR DESCRIPTION
**Pull request by fmdl**
Description of the issue/feature this PR addresses:
In large database with lot of users (e-commerce database), the browser don't respond.

Because all users are loaded.

@tde-banana-odoo

image

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by fmdl** [comment]
OPW #2982658

**Review by rdeodoo — APPROVED** [review]
LGTM but if you create an opw, link it in the commit message `opw-xxx`, not on the github thread ^^

**Comment by tde-banana-odoo** [comment]
Isn't it possible to give a limit instead ?

**Comment by fmdl** [comment]
@tde-banana-odoo you should put a limit here : https://github.com/odoo/odoo/blob/14.0/addons/website_form/static/src/snippets/s_website_form/options.js#L49

**Comment by rdeodoo** [comment]
I doubt a limit is really going to help instead of the extra domain?
What is the actual internal users are outside of this limit? The only available choices will be non internal ones.
I think the current fix is right, it aligns this field domain to the domain already set in the backend on a task that also filter out the share users

**Comment by tde-banana-odoo** [comment]
It is not a fix, it is hiding the bug. If you have a lot of internal users you will get the same issue.

**Comment by rdeodoo** [comment]
That's correct, even if the bug probably only really happens when there is a lot of external users, like big e-commerce as reported.

 Both should be done then IMO I guess, showing external users as Salesperson is not really relevant, but that would be a (acceptable IMO) functional change in stable.
 
 Anyway, I'll leave that up to what you prefer ;-)

**Comment by fmdl** [comment]
The issue with limit is the same if you add an existing fields, in the form, with lot of record. Odoo will load all the records for the selection.

**Comment by fmdl** [comment]
ping @tde-banana-odoo

**Comment by fla-subteno-it** [comment]
ping @rdeodoo

**Comment by tde-banana-odoo** [comment]
Technically we could effectively return sales people, which could probably remove the need of a limit. Anyway other people cannot access lead model... but in case of custom ACLs this would not always be true.

Long story short, I would do as pointed out by @rdeodoo , share domain + a limit :) .

**Review by rdeodoo — APPROVED** [review]
@robodoo r+

On top of being what it should have been since its implementation (as its mimicking the domain in the backend field and as TDE also mentioned it doesn't makes sense to set portal users as a salesperson), this should already be fixing most of the issues, as most likely the "too many users" is coming from the fact that we list all the portal ones.

There might still be the case of a DB with a lot of internal users which would make the browser crash, but it's unlikely (would require multiple thousand of internal users). At least, it is a very marginal part of the DBs.

About the limit, I think it should be done in master and come with a way to filter the results through an input search, because right now it's a simple select. It means that if the record you are looking for is not part of the returned list, you won't see it ever. That would/could be a blocking point.
That issue would not be an issue if the select could be filtered by typing stuff (like in backend fields).

I am fine with adding the limit in stable if you (TDE) judged it really useful, but I have no clue what a good limit would be, as it would need to:
- Prevent the browser crash (I guess it depends a lot on the browser, we can probably assume 1.000?)
- But still avoid hiding records that would be the one the user wants to select

At the same time, is anyone going to complain because he can't find his record in a list of 1000 records? So I guess the second point is actually a non issue.
I guess it would act like a _"developer probably wrongly dev the option and was not precise enough in the domain as thousand records being listed shouldn't be a thing, so let's limit it in the meantime"_ wall

In the meantime, I'll merge this domain as this will also fix opw-3077779
Let me know what limit number you would have put and I'll create a commit for that, I can't push in this PR anyway and I need it merged.

Note that this won't impact already existing form that would rely on a portal sales person (if it even exists) as this field/value will remain set on the form unless the salesperson is explicitly change in the right panel. Any other option modification (like the Sales Team) won't even change that Salesperson ID.

**Comment by tde-banana-odoo** [comment]
Hello,

That's kind but we actually have ongoing bugfixes for website crm form. Please ping / ask before merging in crm as it impacts ongoing tasks.

Cheers !

**Pull request closed without merging**
